### PR TITLE
[app-compat helm-chart] - added service-monitor.yaml

### PR DESCRIPTION
**Pull request by namannagar89**
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the backstage documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: charts/app-compat/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  labels:
+    app: {{ template "application.name" . }}
+spec:
+  endpoints:
+    - port: metrics
+      {{- if .Values.metrics.serviceMonitor.interval }}
+      interval: {{ .Values.metrics.serviceMonitor.interval }}
+      {{- end }}
+      {{- if .Values.metrics.serviceMonitor.scrapeTimeout }}
+      scrapeTimeout: {{ .Values.metrics.serviceMonitor.scrapeTimeout }}
```
Comment (ankit-ls) [inline charts/app-compat/templates/servicemonitor.yaml:16]:
```suggestion
      {{- if .Values.metrics.interval }}
      interval: {{ .Values.metrics.interval }}
      {{- end }}
      {{- if .Values.metrics.scrapeTimeout }}
      scrapeTimeout: {{ .Values.metrics.scrapeTimeout }}
```

File: charts/app-compat/values.yaml [hunk 2]
```diff
@@ -389,6 +389,12 @@ StartupProbe:
 prometheus:
   release: monitoring
 
+metrics:
+  enabled: false
+  port: metrics
+  interval: 10s
```
Comment (ankit-ls) [inline charts/app-compat/values.yaml:395]:
```suggestion
  interval: 30s
```

File: charts/app-compat/values.yaml [hunk 3]
```diff
@@ -389,6 +389,12 @@ StartupProbe:
...
+  scrapeTimeout: 20s
```
Comment (ankit-ls) [inline charts/app-compat/values.yaml:396]:
```suggestion
  scrapeTimeout: 10s
```

File: charts/app-compat/templates/servicemonitor.yaml (hunk 1 above)
Comment (namannagar89) [reply charts/app-compat/templates/servicemonitor.yaml]: done

File: charts/app-compat/values.yaml (hunk 2 above)
Comment (namannagar89) [reply charts/app-compat/values.yaml]: done

Comment (namannagar89) [reply charts/app-compat/values.yaml]: done

File: charts/app-compat/templates/servicemonitor.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{{- if and .Values.metrics.enabled }}
```
Comment (ankit-ls) [inline charts/app-compat/templates/servicemonitor.yaml:1]:
```suggestion
{{- if .Values.metrics.enabled }}
```

**Review by ankit-ls — APPROVED** [review]

**Pull request merged**